dbt/adapters/duckdb/plugins/delta.py

```diff
@@ -0,0 +1,55 @@
+from typing import Any
+from typing import Dict
+
+from deltalake import DeltaTable
+
+from . import BasePlugin
+from ..utils import SourceConfig
+from dbt.logger import GLOBAL_LOGGER as logger
+
+
+class Plugin(BasePlugin):
+    def initialize(self, config: Dict[str, Any]):
+        self._REGISTERED_DF: dict = {}
+    
+    def configure_cursor(self, cursor):
+        for source_table_name, df in self._REGISTERED_DF.items():
+            df_name = source_table_name.replace(".", "_") + "_df"
+            cursor.register(df_name, df)
+            cursor.execute(
+                f"CREATE OR REPLACE VIEW {source_table_name} AS SELECT * FROM {df_name}"
```

Comment:
curious, why the indirection instead of just referencing `df` directly in the create view statement?

Reply:
There are two reasons
1. Historically code was differently structured and I was fighting to define dataframe which will persist between sessions executed by the dbt and therefore had first to  register it in another scope and then to reference it by unique name
2. I was not sure how duckdb registers that dataframe when it is defined in the create statement and especially when you have more then two tables and start to execute things in parallel so i thought unique name per source dataframe makes sense